src/index.ts

```diff
@@ -46,20 +47,29 @@ calypsoServer.get('/localimages', (req: any, res: any) => {
 });
 
 calypsoServer.use(determineCommitHash);
-calypsoServer.get('/status', async (req: any, res: any) => {
+calypsoServer.get('/status', async (req: express.Request, res: express.Response) => {
 	const commitHash = req.session.commitHash;
 	let status;
-	if (isContainerRunning(commitHash)) {
-		status = 'Ready';
-	} else if (await hasHashLocally(commitHash)) {
-		status = 'NeedsPriming';
-	} else if (await isBuildInProgress(commitHash)) {
-		status = 'Building';
-	} else {
-		status = 'NotBuilt';
+	const send = ( m: string ) => ( res.send( m ), res.end() );
+
+	const isRunning = isContainerRunning(commitHash);
+	if ( isRunning ) {
+		return send( 'Ready' );
 	}
-	res.send(status);
-	res.end();
+
+	if ( ! lastBuildFinished.get( commitHash ) ) {
```

Comment:
would this ever reach `Building` state? 
I think we'd be better served with an `isBuildErrored` function that scans the build's logfile for the word "Error"

Reply:
good point - checking for an explicit `false` would be what we really want